0.5.2: pytest fails in `tests/test_find_dependencies.py::test_single_dependency_collect_only` unit

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix> using `installer` module
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

<details>
<summary>Here is pytest output:</summary>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-find-dependencies-0.5.2-2.fc36.x86_64/usr/lib64/python3.9/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-find-dependencies-0.5.2-2.fc36.x86_64/usr/lib/python3.9/site-packages
+ /usr/bin/pytest -ra -m 'not network'
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.9.18, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-find-dependencies-0.5.2
configfile: setup.cfg
testpaths: tests
plugins: find-dependencies-0.5.2
collected 26 items

tests/test_find_dependencies.py ....F.................s..s                                                                                                                            [100%]

========================================================================================= FAILURES ==========================================================================================
____________________________________________________________________________ test_single_dependency_collect_only ____________________________________________________________________________

test_path = <Testdir local('/tmp/pytest-of-tkloczko/pytest-21/test_single_dependency_collect_only0')>

    def test_single_dependency_collect_only(test_path):
        test_path.makepyfile(
            test_one="""
            flag = True
            def test_a(): pass
            def test_b(): assert flag
            def test_c(): global flag; flag = False
            """
        )

        result = test_path.runpytest("--find-dependencies", "--collect-only",
                                     "-p", "no:randomly")
        assert int(result.ret) == 0
>       result.stdout.fnmatch_lines([
            "collected 3 items",
            "<Module test_one.py>",
            "  <Function test_a>",
            "  <Function test_b>",
            "  <Function test_c>"
        ])
E       Failed: nomatch: 'collected 3 items'
E           and: '============================= test session starts =============================='
E           and: 'platform linux -- Python 3.9.18, pytest-8.1.1, pluggy-1.4.0'
E           and: 'rootdir: /tmp/pytest-of-tkloczko/pytest-21/test_single_dependency_collect_only0'
E           and: 'configfile: pytest.ini'
E           and: 'plugins: find-dependencies-0.5.2'
E       exact match: 'collected 3 items'
E       nomatch: '<Module test_one.py>'
E           and: ''
E           and: '<Dir test_single_dependency_collect_only0>'
E           and: '  <Module test_one.py>'
E           and: '    <Function test_a>'
E           and: '    <Function test_b>'
E           and: '    <Function test_c>'
E           and: ''
E           and: '========================== 3 tests collected in 0.00s =========================='
E       remains unmatched: '<Module test_one.py>'

/home/tkloczko/rpmbuild/BUILD/pytest-find-dependencies-0.5.2/tests/test_find_dependencies.py:104: Failed
----------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.9.18, pytest-8.1.1, pluggy-1.4.0
rootdir: /tmp/pytest-of-tkloczko/pytest-21/test_single_dependency_collect_only0
configfile: pytest.ini
plugins: find-dependencies-0.5.2
collected 3 items

<Dir test_single_dependency_collect_only0>
  <Module test_one.py>
    <Function test_a>
    <Function test_b>
    <Function test_c>

========================== 3 tests collected in 0.00s ==========================
================================================================================== short test summary info ==================================================================================
SKIPPED [1] tests/test_find_dependencies.py:543: pytest-xdist not installed
SKIPPED [1] tests/test_find_dependencies.py:613: pytest-xdist not installed
FAILED tests/test_find_dependencies.py::test_single_dependency_collect_only - Failed: nomatch: 'collected 3 items'
========================================================================= 1 failed, 23 passed, 2 skipped in 10.23s ==========================================================================
```
</details>

<details>
<summary>List of installed modules in build env:</summary>

```console
Package            Version
------------------ -----------
build              1.1.1
distro             1.9.0
exceptiongroup     1.1.3
importlib_metadata 7.0.1
iniconfig          2.0.0
installer          0.7.0
packaging          24.0
pluggy             1.4.0
pyproject_hooks    1.0.0
pytest             8.1.1
python-dateutil    2.9.0.post0
setuptools         69.1.1
tokenize_rt        5.2.0
tomli              2.0.1
wheel              0.43.0
zipp               3.17.0
```
</details>

Please let me know if you need more details or want me to perform some diagnostics.

--- a/.github/workflows/pythontests.yml
+++ b/.github/workflows/pythontests.yml
@@ -40,10 +40,14 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest]
-        python-version: [3.7, 3.8, 3.9, "3.10", "3.11", "pypy-3.7"]
-        exclude:
-          - os: windows-latest
-            python-version: pypy3
+        python-version: [3.7, 3.8, 3.9, "3.10", "3.11", "3.12"]
+        include:
+          - python-version: "pypy-3.7"
+            os: ubuntu-latest
+          - python-version: "pypy-3.9"
+            os: ubuntu-latest
+          - python-version: "pypy-3.10"
+            os: ubuntu-latest
     env:
       OS: ${{ matrix.os }}
     steps:
@@ -68,7 +72,7 @@ jobs:
         python -m pip install pytest pytest-cov pytest-randomly tox
     - name: Run tests
       run: |
-        TOX_PYTHON_VERSION=$(if [ ${{ matrix.python-version }} = pypy-3.7 ]; then echo "pypy3"; else echo py${{ matrix.python-version }} | tr -d .; fi)
+        TOX_PYTHON_VERSION=$(if [[ ${{ matrix.python-version }} == pypy* ]]; then echo ${{ matrix.python-version }} | tr -d .-; else echo py${{ matrix.python-version }} | tr -d .-; fi)
         COV_CMD=$(if [ ${{ matrix.python-version }} = 3.10 ]; then echo "--cov=find_dependencies --cov-report=xml"; else echo ; fi) tox -e $(tox -l | grep $TOX_PYTHON_VERSION | paste -sd "," -)
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v4

--- a/setup.py
+++ b/setup.py
@@ -37,6 +37,7 @@ setup(
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
         'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: Implementation :: CPython',
         'Programming Language :: Python :: Implementation :: PyPy',

--- a/tests/test_find_dependencies.py
+++ b/tests/test_find_dependencies.py
@@ -103,10 +103,10 @@ def test_single_dependency_collect_only(test_path):
     assert int(result.ret) == 0
     result.stdout.fnmatch_lines([
         "collected 3 items",
-        "<Module test_one.py>",
-        "  <Function test_a>",
-        "  <Function test_b>",
-        "  <Function test_c>"
+        "*<Module test_one.py>",
+        "*  <Function test_a>",
+        "*  <Function test_b>",
+        "*  <Function test_c>"
     ])
 
 

--- a/tox.ini
+++ b/tox.ini
@@ -1,22 +1,22 @@
 # For more information about tox, see https://tox.readthedocs.io/en/latest/
 [tox]
 envlist =
-    {py37,py38,py39,pypy3}-pytest{43,50,54,60,61,62,70,72}
-    {py310,py311}-pytest{624,70,72}
+    {py37,pypy37}-pytest{50,54,60,62,624,70,74}
+    {py38,py39,pypy39}-pytest{50,54,60,62,624,70,74,80}
+    {py310,py311,py312,pypy310}-pytest{624,70,74,80}
 
 [testenv]
 deps =
-    pytest43: pytest>=4.3,<4.4
     pytest50: pytest>=5.0,<5.1
     pytest54: pytest>=5.4,<6.0
     pytest60: pytest>=6.0,<6.1
-    pytest61: pytest>=6.1,<6.2
     pytest62: pytest>=6.2,<6.3
     pytest624: pytest>=6.2.4,<6.3
     pytest70: pytest>=7.0,<7.1
-    pytest72: pytest>=7.2,<7.3
-    pytest{43,50,54}: pytest-xdist<2.0.0
-    pytest{60,61,62,624,70,72}: pytest-xdist
+    pytest74: pytest>=7.3,<8.0
+    pytest80: pytest==8.0.1
+    pytest{50,54}: pytest-xdist<2.0.0
+    pytest{60,62,624,70,74,80}: pytest-xdist
     pytest-cov
     pytest-randomly
 

